modules/git/command.go

```diff
@@ -148,13 +150,21 @@ func (c *Command) Run(opts *RunOpts) error {
 		cmd.Env = opts.Env
 	}
 
+	if GlobalConfigFile == "" {
+		// TODO: now, some unit test code call the git module directly without initialization, which is incorrect.
+		// at the moment, we just use a temp gitconfig to prevent from conflicting with user's gitconfig
+		// in future, the git module should be initialized first before use.
+		GlobalConfigFile = filepath.Join(os.TempDir(), "/gitea-temp-gitconfig")
+		log.Warn("GlobalConfigFile is empty, the git module is not initialized correctly, using a temp gitconfig (%s) temporarily", GlobalConfigFile)
+	}
+
 	cmd.Env = append(
 		cmd.Env,
 		fmt.Sprintf("LC_ALL=%s", DefaultLocale),
-		// avoid prompting for credentials interactively, supported since git v2.3
-		"GIT_TERMINAL_PROMPT=0",
-		// ignore replace references (https://git-scm.com/docs/git-replace)
-		"GIT_NO_REPLACE_OBJECTS=1",
+		"GIT_TERMINAL_PROMPT=0",               // avoid prompting for credentials interactively, supported since git v2.3
+		"GIT_NO_REPLACE_OBJECTS=1",            // ignore replace references (https://git-scm.com/docs/git-replace)
+		"GIT_CONFIG_NOSYSTEM=1",               // https://git-scm.com/docs/git-config
+		"GIT_CONFIG_GLOBAL="+GlobalConfigFile, // make Gitea use internal git config only, to prevent conflicts with user's git config
```

Comment:
Looks like this ask git version 2.32
https://github.com/git/git/commit/4179b4897f2de28858acaebd6382c06c91532e98

For old git version, an instead method is to set `HOME` and `XDG_CONFIG_HOME` to $RepoRoot in the git command process.

Another problem is for go-git library, I don't know if the library will read the git config setting.

Reply:
After the CI passes, will add these support.

Reply:
After some struggling, I decide to use `HOME=xxx` instead of `GIT_CONFIG_GLOBAL`

The reason is that git always requires a home directory, and XDG_CONFIG_HOME's policy is too complex to handle.